### PR TITLE
fix: modified send_email_handler conditional statement

### PR DESCRIPTION
**Pull request by jcomrade**

### CODE REVIEW

**Review by ArJSarmiento — CHANGES_REQUESTED** [review]
Good idea! Pls implement skipping this way. Thanks

File: backend/usecase/email_usecase.py [hunk 1]
```diff
@@ -28,8 +28,8 @@ def __init__(self) -> None:
         }
         self.__event_email = None
 
-    def __send_email_handler(self, email_in_list: List[EmailIn]) -> Tuple[HTTPStatus, str]:
-        if self.__event_email and self.__event_email == SpecialEmails.AWSUG_DAVAO.value:
+    def __send_email_handler(self, email_in_list: List[EmailIn], event: Event) -> Tuple[HTTPStatus, str]:
+        if self.__event_email and event.konfhubId and event.konfhubApiKey:
```
Comment (ArJSarmiento) [inline backend/usecase/email_usecase.py:32]: Add a comment why this check exists

File: backend/usecase/email_usecase.py [hunk 2]
```diff
@@ -28,8 +28,8 @@ def __init__(self) -> None:
         }
         self.__event_email = None
 
-    def __send_email_handler(self, email_in_list: List[EmailIn]) -> Tuple[HTTPStatus, str]:
-        if self.__event_email and self.__event_email == SpecialEmails.AWSUG_DAVAO.value:
+    def __send_email_handler(self, email_in_list: List[EmailIn], event: Event) -> Tuple[HTTPStatus, str]:
```
Comment (ArJSarmiento) [inline backend/usecase/email_usecase.py:31]: Add a docstring for the input and expected output

File: backend/usecase/email_usecase.py
Comment (ArJSarmiento) [inline backend/usecase/email_usecase.py:1]: Some `send_batch_email` calls were not refactored. Please check for all instances.

**Review by ArJSarmiento — CHANGES_REQUESTED** [review]
Please check PR comments. Thanks!

File: backend/usecase/email_usecase.py [hunk 3]
```diff
@@ -375,4 +395,4 @@ def send_event_completion_email(
             for participant in participants
         ]
 
-        return self.send_batch_email(email_in_list=emails)
+        return self.send_batch_email(email_in_list=emails, event=Event)
```
Comment (ArJSarmiento) [inline backend/usecase/email_usecase.py:398]: should be the event variable not the type

File: backend/usecase/email_usecase.py [hunk 4]
```diff
@@ -194,7 +214,7 @@ def send_accept_reject_status_email(
                 preregistration_entry=preregistration, preregistration_in=PreRegistrationPatch(acceptanceEmailSent=True)
             )
 
-        return self.send_batch_email(email_in_list=emails)
+        return self.send_batch_email(email_in_list=emails, event=Event)
```
Comment (ArJSarmiento) [inline backend/usecase/email_usecase.py:217]: should be the event variable not the type. Pls check for the rest of the instances

**Review by ArJSarmiento — CHANGES_REQUESTED** [review]
Please check comments. Thanks!

**Review by ArJSarmiento — APPROVED** [review]
LGTM. Thanks!

**Comment by jcomrade** [comment]
[konfhubSample.webm](https://github.com/user-attachments/assets/11381168-aecb-4396-b4e9-b09b85d77d2e)

**Comment by jcomrade** [comment]
[NO_Konfhub_credentials.webm](https://github.com/user-attachments/assets/832ed552-3721-4812-8870-e88cfd5a73c3)

Event Creation and pre-registration without  konfhub id and apikey

Expected output: Email Received for Event Creation

**Pull request merged**